### DISCUSSION TITLE
`Menu` and `MenuBar` - what is `HandleGrabView` supposed to do?

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#3289] [Q&A] [answered]
Looking at both `develop` and `v2_develop` I can't figure out why menu feels it needs to grab the mouse.

I've commented out all places where it calls `Application.GrabMouse` and `MenuBar`, `Menu`, and `ContextMenu` all seem to behave the same.

@BDisp do you recall why menu wanted to grab the mouse?

**Comment by BDisp** [comment]
Because when the `MenuBar` is open and clicking the mouse outside it, will close the `MenuBar`, otherwise it will never be closed if not grab the mouse.

**Reply by tig** [reply]
Ahh. Makes sense.

But what's all the other logic for? In particular, I'm trying to figure what causes this code to execute (and what it's trying to do):

```cs
if (me.Y > -1)
{
    Point newxy = v.ScreenToFrame (me.X, me.Y);

    nme = new MouseEvent
    {
        X = newxy.X,
        Y = newxy.Y,
        Flags = me.Flags,
        OffsetFromScreenX = me.X - newxy.X,
        OffsetFromScreenY = me.Y - newxy.Y,
        View = v
    };
}
else
{
     nme = new MouseEvent { X = me.X + current.Frame.X, Y = 0, Flags = me.Flags, View = v };
}
v.OnMouseEvent (nme);
```

I've set a breakpoint on `v.OnMouseEvent (nme)` and can't get it to hit with any scenario or unit test.

**Reply by BDisp** [reply]
That already is some modified code because on v1 the `ScreenToFrame` didn't exist.

**Reply by tig** [reply]
I understand it's modified code. 

I was looking at the v1 source as well and I can't see any reason for Menu to care about `OfX/OfY`. 

If nobody can say why it's needed, I'm just going to remove the code.

**Reply by BDisp** [reply, marked as answer]
I can't answer if it has or not impact now. Remove and see if it really isn't needed. If it behaves badly then you will see it's really necessary. I can't answer that because I'm not very sure about it.

**Comment by tig** [comment]
I've removed the related dead code in

- https://github.com/gui-cs/Terminal.Gui/pull/3278
